src/NServiceBus.Core.Tests/Routing/SubscriptionRouterTests.cs

```diff
@@ -31,18 +32,18 @@ public void Should_return_correct_base_and_inherited_address_even_if_they_differ
             publishers.AddStatic(inheritedEndpoint, baseType);
             publishers.AddStatic(inheritedEndpoint, inheritedType );
             var knownEndpoints = new EndpointInstances();
-            knownEndpoints.AddDynamic(e => new [] { new EndpointInstance(e, null, null) });
+            knownEndpoints.AddDynamic(e => Task.FromResult(EnumerableEx.Single(new EndpointInstance(e, null, null))));
```

Comment:
Why not just `knownEndpoints.AddDynamic(e => Task.FromResult(new [] { new EndpointInstance(e, null, null) }));`?